core/src/main/scala/org/ensime/core/Project.scala

```diff
@@ -12,10 +12,13 @@ import org.ensime.indexer._
 
 import scala.collection.immutable.ListSet
 import scala.concurrent.duration._
+import scala.util.Properties._
 import scala.util._
 import org.ensime.util.file._
 import org.ensime.util.FileUtils
 
+case object CloseOnIndexUpdate
```

Comment:
There is no idomatic way as far as I'm aware - I don't think poison pilling the actor does not give the behaviour you want.